src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.MaxNumber.cs

```diff
@@ -87,20 +88,30 @@ public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> y)
             {
                 if (typeof(T) == typeof(float) || typeof(T) == typeof(double))
                 {
-                    if (AdvSimd.IsSupported && typeof(T) == typeof(float))
+                    // We can't use AdvSimd.MaxNumber here because it doesn't correctly
+                    // handle sNaN (it converts it to qNaN as per the now deprecated
+                    // maxNum function defined by IEEE 754:2008, but which is not inline
+                    // with the maximumNumber function that replaces it in IEEE 754:2019
+
+                    Vector128<T> max;
+
+                    if (Sse.IsSupported && typeof(T) == typeof(float))
```

Comment:
Nit: presumably this change is unrelated to the test fix at hand, would it be possible to update the PR title/commit message to include this change?

Reply:
It's not a change. It's preserving the behavior and perf for xarch from prior to this PR

Reply:
That is, `Vector128.Min` on xarch currently emits `minps`/`minpd` (`Sse.Min` for float and `Sse2.Min` for double) while on Arm64 it emits `fmin`.

We therefore cannot use `Vector128.Min` on Arm64 since it propagates NaN (and longterm wouldn't work for xarch either, as we're working towards ensuring consistency there). But, it's safe to continue using the underlying intrsinsics for xarch and thus allows us to preserve codegen as-is there.